Server/Jobs/DeleteCrashReportDumpJob.cs

```diff
@@ -37,8 +37,7 @@ public class DeleteCrashReportDumpJob
 
         var filePath = Path.Combine(baseFolder, report.DumpLocalFileName);
 
-        await semaphore.WaitAsync(cancellationToken);
-        try
+        using (await fileLocks.LockAsync(baseFolder, cancellationToken).ConfigureAwait(false))
```

Comment:
Could you switch this using to be an using statement (`await using fileLocks.LockAsync(...);`)?

Reply:
Do you mean `using (await fileLocks.LockAsync(baseFolder, cancellationToken).ConfigureAwait(false));`?

Then the release would happen after the logging
`logger.LogInformation("Deleted crash dump file {DumpLocalFileName}", report.DumpLocalFileName);`

Is this what you want?

Reply:
Yeah, I suppose so (though one level of the parentheses might be optional). I kind of missed the detail where you need to await the  `LockAsync` and the result value is not `IAsyncDisposable`.

The release happening after the log is so inconsequential that I usually just keep locks in those cases as it reduces one level of nesting and makes the code look nicer.

Reply:
The code analysis doesn't like this:

`using var _ = await fileLocks.LockAsync(baseFolder, cancellationToken).ConfigureAwait(false);`

I need to assign it an actual variable. So it will have to be something like this:

`using var asyncKeyedLock = await fileLocks.LockAsync(baseFolder, cancellationToken).ConfigureAwait(false);`

Reply:
Ah yeah, of course. The using statement requires a variable name, I forgot that for a second.

Reply:
Do you still want it this way or shall I revert?

Reply:
It's fine, the naming on the variable could be improved, but I did use "semaphore" as the name previously so I can't really fault you for not using a more imaginative name. I just had one more different comment on the style I just posted.

Reply:
This is taking longer than I thought it would :)
Please suggest a new variable name.

Reply:
I'd name these `baseFolderLock` (or whatever the string holding the path is named) as they are the lock you need to hold while using the `baseFolder` variable, that way there's an immediately obvious connection.

Reply:
Like I said above, you can rename them if you like, but I'm fine with accepting them as they are to not be overly nitpicky.

Reply:
You do set a record for nitpicking that I've seen, but that's OK because I appreciate the passion :) Done.

Reply:
I did try to say that you didn't *have* to change it...
I sometimes (or maybe even often) comment on small things that I'm fine with being left as is, which I usually try to clarify doesn't *need* to be changed, but the PR author may want to do them anyway to make the highest possible quality PR.